src/Rules/UseTypedReturnForNewMethodsRule.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/*
+ * Copyright (c) Since 2007 PrestaShop.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace PHPStanForPrestaShop\Rules;
+
+use PhpParser\Node;
+use PhpParser\Node\Stmt\ClassMethod;
+use PHPStan\Analyser\Scope;
+use PHPStan\Rules\Rule;
+use PHPStan\Rules\RuleErrorBuilder;
+use PHPStanForPrestaShop\PHPConfigurationLoader\ConfigurationLoaderInterface;
+
+/**
+ * @implements Rule<Node\Stmt\ClassMethod>
+ */
+class UseTypedReturnForNewMethodsRule implements Rule
+{
+    /** @var array */
+    private $excludedClassMethodsList;
+
+    /**
+     * @param ConfigurationLoaderInterface $configurationFileLoader
+     */
+    public function __construct(ConfigurationLoaderInterface $configurationFileLoader)
+    {
+        $this->excludedClassMethodsList = $configurationFileLoader->load();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getNodeType(): string
+    {
+        return ClassMethod::class;
+    }
+
+    /**
+     * @param ClassMethod $node
+     * @param Scope $scope
+     *
+     * @return array
+     * @throws \PHPStan\ShouldNotHappenException
+     */
+    public function processNode(Node $node, Scope $scope): array
```

Comment:
The logic here follows this line:
1. check if function is part of the exclusion list, if yes, return nothing.
2. check if function has return type, if yes, return nothing.
3. if class has parents, look at them and see if they have the same method ; if no, then it's a rule violation.
4. if class has parents who also have this class method, see if they are part of the exclusion list, if yes, return nothing.
5. return violation else.